src/pages/rental-details/rental-details.ts

```diff
@@ -27,7 +27,14 @@ export class RentalDetailsPage {
     this.submitted = true;
 
     if (form.valid) {
-      this.inventoryData.rent(this.items, this.details).subscribe(
+      let promises = [];
+
+      for (let item of this.items) {
```

Comment:
Any reason to use `let` here instead of `const`? Off the top of my head, they are exactly the same in this situation. I generally like to use `const` unless the variable needs to change after assignment, though, for clarity's sake.